### PR TITLE
feat: theory from LeanSAT

### PR DESCRIPTION
**Pull request by hargoniX**

### CODE REVIEW

File: src/Init/PropLemmas.lean [hunk 1]
```diff
@@ -309,6 +309,11 @@ theorem not_forall_of_exists_not {p : α → Prop} : (∃ x, ¬p x) → ¬∀ x,
 @[simp] theorem exists_eq_right_right' : (∃ (a : α), p a ∧ q a ∧ a' = a) ↔ p a' ∧ q a' := by
   simp [@eq_comm _ a']
 
+@[simp] theorem exists_or_eq_left (y : α) (p : α → Prop) : ∃ x : α, x = y ∨ p x := ⟨y, .inl rfl⟩
```
Comment (TwoFX) [inline src/Init/PropLemmas.lean:312]: Is there a reason why this is not
```suggestion
@[simp] theorem exists_or_eq_left (y : α) (p : Prop) : ∃ x : α, x = y ∨ p := ⟨y, .inl rfl⟩
```

Comment (hargoniX) [reply src/Init/PropLemmas.lean:312]: The current version is able to deal with the case where `p` depends on the existentially bound `x`, your version would not be able to do that, e.g.:
```lean

namespace Foo
theorem exists_or_eq_left (y : α) (p : α → Prop) : ∃ x : α, x = y ∨ p x := ⟨y, .inl rfl⟩

example : ∃ x, x = 0 ∨ 0 ≤ x := by simp only [exists_or_eq_left]

end Foo

namespace Bar

theorem exists_or_eq_left (y : α) (p : Prop) : ∃ x : α, x = y ∨ p := ⟨y, .inl rfl⟩

example : ∃ x, x = 0 ∨ 0 ≤ x := by simp only [exists_or_eq_left]

end Bar
```

Comment (TwoFX) [reply src/Init/PropLemmas.lean:312]: Oh right, I had the "more general" relation the wrong way around in my head :) Thank you for the explanation.

**Comment by jcommelin** [comment]
LGTM, fwiw

**Pull request merged**